datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -365,4 +428,40 @@ mod tests {
         assert_eq!(floats.value(2), 4.0);
         assert_eq!(floats.value(3), 4.0);
     }
+
+    #[test]
+    fn test_round_f32() {
+        let args: Vec<ArrayRef> = vec![
+            Arc::new(Float32Array::from(vec![125.2345; 10])), // input
+            Arc::new(Int64Array::from(vec![0, 1, 2, 3, 4, 5, -1, -2, -3, -4])), // decimal_places
+        ];
+
+        let result = round(&args).expect("failed to initialize function round");
+        let floats =
+            as_float32_array(&result).expect("failed to initialize function round");
+
+        let expected = Float32Array::from(vec![
+            125.0, 125.2, 125.23, 125.235, 125.2345, 125.2345, 130.0, 100.0, 0.0, 0.0,
```

Comment:
I spot checked that these answers are consistent with postgres 👍  (including negative)

```sql
postgres=# select round(125.2345, -1);
 round
-------
   130
(1 row)

postgres=# select round(125.2345, -2);
 round
-------
   100
(1 row)

postgres=# select round(125.2345, 3);
  round
---------
 125.235
(1 row)
```